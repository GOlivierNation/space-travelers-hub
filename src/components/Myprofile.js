import React from 'react';
const Myprofile = () => (
  <div>My Profile</div>
);
export default Myprofile;
