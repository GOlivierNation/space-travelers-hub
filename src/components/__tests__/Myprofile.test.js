import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Myprofile from '../Myprofile';

describe('Test for Myprofile page', () => {
  
  test('The page should renders My Missions', () => {
    render(<Provider store={store}><Myprofile /></Provider>);
    const element = screen.getByText(/My Missions/i);
    expect(element).toBeInTheDocument();
  });

  test('The page should renders My Rockets', () => {
    render(<Provider store={store}><Myprofile /></Provider>);
    const element = screen.getByText(/My Rockets/i);
    expect(element).toBeInTheDocument();
  });

  it('Should render the Myprofile page', () => {
    const tree = renderer.create(<Provider store={store}><Myprofile /></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
