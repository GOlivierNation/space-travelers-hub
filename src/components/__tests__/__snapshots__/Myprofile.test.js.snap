// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Myprofile page Should render the Myprofile page 1`] = `
<div
  className="profile-container"
>
  <div
    className="my-missions"
  >
    <h2>
      My Missions
    </h2>
    <div
      className="my-missions-list"
    />
  </div>
  <div
    className="my-rockets"
  >
    <h2>
      My Rockets
    </h2>
    <div
      className="my-rockets-list"
    />
  </div>
</div>
`;
